document/core/exec/instructions.rst

```diff
@@ -2620,35 +2620,42 @@ Control Instructions
    \end{array}
 
 
-.. _exec-try:
+.. _exec-try-catch:
 
-:math:`\TRY~\blocktype~\instr_1^\ast~\CATCH~\instr_2^\ast~\END`
-...............................................................
-
-1. Assert: due to :ref:`validation <valid-blocktype>`, :math:`\expand_F(\blocktype)` is defined.
-
-2. Let :math:`[t_1^n] \to [t_2^m]` be the :ref:`function type <syntax-functype>` :math:`\expand_F(\blocktype)`.
+:math:`\TRY~\blocktype~\instr_1^\ast~(\CATCH~x~\instr_2^\ast)^\ast~(\CATCHALL~\instr_3^\ast)^?~\END`
+....................................................................................................
 
-3. Assert: due to :ref:`validation <valid-try>`, there are at least :math:`n` values on the top of the stack.
+**TODO: Add prose**
 
-4. Pop the values :math:`\val^n` from the stack.
+.. math::
+   ~\\[-1ex]
+   \begin{array}{l}
+   F; \val^n~(\TRY~\X{bt}~\instr_1^\ast~(\CATCH~x~\instr_2^\ast)^\ast~(\CATCHALL~\instr_3^\ast)^?~\END
+   \quad \stepto \\
+   \qquad F; \LABEL_m\{\}~(\CATCHadm\{a~\instr_2^\ast\}^\ast\{\epsilon~\instr_3\}^?~\val^n~\instr_1^\ast~\END)~\END \\
+   (\iff \expand_F(\X{bt}) = [t_1^n] \to [t_2^m] \land (F.\AMODULE.\MITAGS[x]=a)^\ast)
+   \end{array}
 
-5. Let :math:`L` be the label whose arity is :math:`m` and whose continuation is the end of the |TRY| instruction.
 
-6. Let :math:`H` be the exception handler whose arity is :math:`m` and whose continuation is the beginning of :math:`\instr_2^\ast`.
+.. _exec-try-delegate:
 
-7. :ref:`Enter <exec-handler-enter>` the exception handler `H`.
+:math:`\TRY~\blocktype~\instr^\ast~\DELEGATE~l`
+...............................................
 
-8. :ref:`Enter <exec-instr-seq-enter>` the block :math:`\val^n~\instr_1^\ast` with label :math:`L`.
+**TODO: Add prose**
 
 .. math::
    ~\\[-1ex]
-   \begin{array}{lcl@{\qquad}}
-   F; \val^n~(\TRY~\X{bt}~\instr_1^\ast~\CATCH~\instr_2^\ast~\END &\stepto&
-   \CATCHN_m\{\instr_2\}~(\LABEL_m \{\}~\val^n~\instr_1^\ast~\END)~\END \\
-   \hspace{5ex}(\iff \expand_F(\X{bt}) = [t_1^n] \to [t_2^m]) &&\\
+   \begin{array}{lcl}
+   F; \val^n~(\TRY~\X{bt}~\instr^\ast~\DELEGATE~l) &\stepto&
+   F; \LABEL_m\{\}~(\DELEGATEadm\{l\}~\val^n~\instr^\ast~\END)~\END \\
+   && (\iff \expand_F(\X{bt}) = [t_1^n] \to [t_2^m])
    \end{array}
 
+.. note::
+   Note that the last reduction step above is similar to the :ref:`reduction <exec-br>` of :math:`\BR~l`, but also doing a throw after it breaks.
```

Comment:
This note is misplaced, it's meant to describe the reduction of `delegate_adm`, not that of `try ... delegate l`.

I'm moving this below to the reduction of 'delegate_adm` in the next commit.

To explain why I wrote this:
because I think of `delegate_adm`'s execution step as doing a `br l` and then (re)throwing the exception.
The reduction steps are similar, at least in this form of `delegate_adm` (with the label outside):

The reduction of `br l`:
```
label_n{instr*} B^{l}[ val^n (br l)] end ↪ val^n instr*
```
and the reduction of `delegate_adm`:
```
label_n{} B^{l}[delegate_adm{l} T[(throw a)] end] end ↪ T[(throw a)]
```